src/main/java/gov/cabinetoffice/gap/adminbackend/services/ApplicationFormSectionService.java

```diff
@@ -118,4 +120,26 @@ public void updateSectionStatus(final Integer applicationId, final String sectio
         this.applicationFormRepository.save(applicationForm);
     }
 
+    public void updateSectionTitle(final Integer applicationId, final String sectionId, final String title) {
+        AdminSession session = HelperUtils.getAdminSessionForAuthenticatedUser();
+        ApplicationFormEntity applicationForm = this.applicationFormRepository.findById(applicationId)
+                .orElseThrow(() -> new NotFoundException("Application with id " + applicationId + " does not exist"));
+
+        if (!session.getGrantAdminId().equals(applicationForm.getCreatedBy())) {
+            throw new AccessDeniedException("User " + session.getGrantAdminId()
+                    + " is unable to access the application form with id " + applicationId);
+        }
```

Comment:
If you merge in develop can use the other `applicationFormRepository.findById()` method with the postauth annotation which makes this redundant